apps/balancer-tools/src/lib/gql.ts

```diff
@@ -61,3 +61,9 @@ export const internalBalances = {
   gql: (chainId: string) =>
     balancerSdks[networkFor(chainId)](internalBalances.client(chainId)),
 };
+
+export const blocks = {
+  client: clientFor(Subgraph.ChainsBlocks),
+  gql: (chainId: string) =>
+    balancerSdks[networkFor(chainId)](blocks.client(chainId)),
+};
```

Comment:
@ribeirojose I'm not sure about the blocks strategy. I touched on this before but I don't remember if we don't want to use subgraphs or not. Decided to use it here to make this validation easier

Reply:
some networks have consistent block times e.g. Ethereum 13s, Gnosis 5s, etc, so for those you may not need a Subgraph, take a look at the [SDK](https://github.com/search?q=repo:balancer/balancer-sdk%20blockTime&type=code) for a reference

Reply:
hmmmm got it but the way we build the gql service it will make the files for all the chains no matter what. would we prefer to have something like this to not use  unnecessary computational power?

Reply:
we can take a look this later on